### ISSUE TITLE
deployment-files fails on integers and booleans

### ISSUE THREAD

**Issue by yannickvr** [aws-cloudformation/cfn-lint#4240]
### CloudFormation Lint Version

1.39.1

### What operating system are you using?

Mac, ubuntu

### Describe the bug

When linting using the `--deployment-files` feature, linting fails with a `E0100 Deployment file 'my-sync-file.yaml' is not supported` error.

After a bit of testing, i found that the parameters can only contain strings. E.g. this works:

```yaml
template-file-path: ../my-template.yml
parameters:
  AppName: my-service
tags:
  Environment: dev
```

This gives the error:

```yaml
template-file-path: ../my-template.yml
parameters:
  AppName: my-service
  CpuUnits: 4096
tags:
  Environment: dev
```

This as well:

```yaml
template-file-path: ../my-template.yml
parameters:
  Public: true
tags:
  Environment: dev
```

### Expected behavior

integers and booleans are processed correctly

### Reproduction template

```yaml
template-file-path: ../my-template.yml
parameters:
  Public: true
  Numer: 666
tags:
  Environment: dev
```
